test/SpokePool.Relay.ts

```diff
@@ -175,199 +175,220 @@ describe("SpokePool Relayer Logic", async function () {
     ).to.be.revertedWith("relay filled");
   });
   it("Can signal to relayer to use updated fee", async function () {
-    const spokePoolChainId = await spokePool.chainId();
-    const { signature } = await modifyRelayHelper(
-      consts.modifiedRelayerFeePct,
-      consts.firstDepositId.toString(),
-      spokePoolChainId.toString(),
-      depositor
-    );
+    await testUpdatedFeeSignaling(depositor.address);
+  });
+  it("EIP1271 - Can signal to relayer to use updated fee", async function () {
+    await testUpdatedFeeSignaling(erc1271.address);
+  });
+  it("Can fill relay with updated fee by including proof of depositor's agreement", async function () {
+    await testFillRelayWithUpdatedFee(depositor.address);
+  });
+  it("EIP1271 - Can fill relay with updated fee by including proof of depositor's agreement", async function () {
+    await testFillRelayWithUpdatedFee(erc1271.address);
+  });
+  it("Updating relayer fee signature verification failure cases", async function () {
+    await testUpdatedFeeSignatureFailCases(depositor.address);
+  });
+  it("EIP1271 - Updating relayer fee signature verification failure cases", async function () {
+    await testUpdatedFeeSignatureFailCases(erc1271.address);
```

Comment:
Can you add a test case for the `erc1271` contract in particular that tries to re-enter the SpokePool via a malicious `isValidSignature` function? This way we can confirm that the SpokePool is protected against reentrancy via this avenue

Reply:
Hm actually maybe we should add a comment that reentrancy might be impossible:

"https://github.com/across-protocol/contracts-v2/commit/6e69fdcef92eb9867409e3fd58387aaef73b81e1#diff-c828c513da79ba8aaef90514612391ab755989974a14d8e480ae02ebd9cfb4beL683"

Reply:
Since isValidSignature is a view method, re-entrancy can happen, but it cannot affect state.

Reply:
Thanks for the input! I added them [here](https://github.com/across-protocol/contracts-v2/pull/221/commits/15cdaf60828edd6e2c56f3ef8c2fcdae31cfedd3)